auth.go

```diff
@@ -146,6 +146,13 @@ func (a Authenticator) Token(state string, r *http.Request) (*oauth2.Token, erro
 	return a.config.Exchange(a.context, code)
 }
 
+// TokenwithoutRequest pulls an authorization code from an HTTP request and attempts to exchange
```

Comment:
Just a few comments:

1. Please update the godoc.  This doesn't _pull an authorization code from an HTTP request_ anymore.
2. I would prefer to still pass in the `state` and do a comparison internally.  It's too easy to forget otherwise and open yourself up to XSRF attacks.